iamidentityv1/iam_identity_v1_integration_test.go

```diff
@@ -1340,6 +1346,103 @@ var _ = Describe(`IamIdentityV1 Integration Tests`, func() {
 			fmt.Fprintf(GinkgoWriter, "UpdateAccountSettings response:\n%s\n", common.ToJSON(accountSettingsResponse))
 		})
 	})
+    
+	Describe(`CreateInactivityReport - Create an inactivity report`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It(`CreateReport(createReportOptions *CreateReportOptions)`, func() {
+
+			createReportOptions := &iamidentityv1.CreateReportOptions{
+				AccountID:   &accountID,
+				Type:        core.StringPtr("inactive"),
+				Duration:    core.StringPtr("120"),
+			}
+
+			reportRef, response, err := iamIdentityService.CreateReport(createReportOptions)
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(202))
+			Expect(reportRef).ToNot(BeNil())
+			fmt.Fprintf(GinkgoWriter, "CreateReport response:\n%s\n", common.ToJSON(reportRef))
+
+			reportId = *reportRef.Reference
+			Expect(reportId).ToNot(BeNil())
+		})
+	})
+    
+	Describe(`GetInactivityReportIncomplete - Get an incomplete inactivity report`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+        It(`GetReport(getReportOptions *GetReportOptions)`, func() {
+            Expect(reportId).ToNot(BeEmpty())
+            getReportOptions := &iamidentityv1.GetReportOptions{
+              AccountID: &accountID,
+              Reference: &reportId,
+            }
+
+            report, response, err := iamIdentityService.GetReport(getReportOptions)
+
+            Expect(err).To(BeNil())
+            Expect(response.StatusCode).To(Equal(204))
+            Expect(report).To(BeNil())
+		})
+	})
+
+	Describe(`GetInactivityReportComplete - Get a complete inactivity report`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It(`GetReport(getReportOptions *GetReportOptions)`, func() {
+			Expect(reportId).ToNot(BeEmpty())
+			getReportOptions := &iamidentityv1.GetReportOptions{
+				AccountID: &accountID,
+				Reference: &reportId,
+			}
+            
+            for {
```

Comment:
It looks like the logic here (with the "infinite" for loop) is to invoke the operation until it returns a status code other than 204. How do we know when/if we'll ever get a non-204 status code back???   This almost seems like a "paginated" response but there's no pagination :)

Reply:
We are generating a report and while report generation is in progress it returns a 204. At some point in time the report will be either complete (and thus return a 200) or failed and return an error. But I accept that we should have a defensive strategy and stop looping after a while. Will improve...